src/Mapping/DataTarget/Direct.php

```diff
@@ -152,12 +152,20 @@ protected function checkAssignData($newData, $valueContainer, $getter)
             $fieldName = $fieldNameParts[2];
         }
 
-        $fieldDefinition = $this->getFieldDefinition($valueContainer, $fieldName);
-        if ($this->writeIfTargetIsNotEmpty === false && !$fieldDefinition->isEmpty($currentData)) {
+        try {
+            $fieldDefinition = $this->getFieldDefinition($valueContainer, $fieldName);
+            $currentDataIsEmpty = $fieldDefinition->isEmpty($currentData);
+            $newDataIsEmpty = $fieldDefinition->isEmpty($newData);
+        } catch (InvalidConfigurationException $e) {
+            $currentDataIsEmpty = empty($currentData);
+            $newDataIsEmpty = empty($newData);
+        }
```

Comment:
```suggestion
        if ($this->fieldName === 'key'){
            $currentDataIsEmpty = empty($currentData);
            $newDataIsEmpty = empty($newData);
        } else {
            $fieldDefinition = $this->getFieldDefinition($valueContainer, $fieldName);
            $currentDataIsEmpty = $fieldDefinition->isEmpty($currentData);
            $newDataIsEmpty = $fieldDefinition->isEmpty($newData);
        }
```

maybe it's more safer using this?